File: GameLibrary/test/cc/lib/utils/GridTest.java
package cc.lib.utils;

import junit.framework.TestCase;

public class GridTest extends TestCase {

    public void testPosIndex() {

        for (int i=0; i<2<<15; i++) {
            for (int ii=0; ii<2<<15; ii++) {
                Grid.Pos p = new Grid.Pos(i, ii);
                int index = p.getIndex();
                Grid.Pos t = Grid.Pos.fromIndex(index);
                assertEquals(p, t);
            }
        }

    }

    public void testEnsureCapacity() {

        Grid<String> grid = new Grid(2, 2);
        System.out.println(grid.toStringNumbered());

        grid.fill("hello");
        System.out.println(grid.toStringNumbered());

        grid.ensureCapacity(3, 3, "Goodbye");
        System.out.println(grid.toStringNumbered());

    }
}

File: GameLibrary2/test/cc/lib/utils/GridTest.kt
package cc.lib.utils

import cc.lib.utils.MirroredGrid.Pos.Companion.fromIndex
import junit.framework.TestCase

class GridTest : TestCase() {
	fun testPosIndex() {
		for (i in 0 until (2 shl 15)) {
			for (ii in 0 until (2 shl 15)) {
				val p = MirroredGrid.Pos(i, ii)
				val index = p.index
				val t = fromIndex(index)
				assertEquals(p, t)
			}
		}
	}

	fun testEnsureCapacity() {
		val grid: MirroredGrid<String> = newMirroredGrid(2, 2) { _, _ -> "" }
		grid.fill("hello")
		grid.ensureCapacity(3, 3) { _, _ -> "Goodbye" }
		println(grid.toString())
	}
}
